robolectric-shadows/shadows-core/src/main/resources/org/robolectric/fakes/ProjectCallback.java.vm

```diff
@@ -0,0 +1,97 @@
+package org.robolectric.fakes;
+#if ($api >= 21)
+import com.android.ide.common.rendering.api.ActionBarCallback;
+import com.android.ide.common.rendering.api.AdapterBinding;
+import com.android.ide.common.rendering.api.ILayoutPullParser;
+import com.android.ide.common.rendering.api.IProjectCallback;
+import com.android.ide.common.rendering.api.ResourceReference;
+import com.android.ide.common.rendering.api.ResourceValue;
+import com.android.resources.ResourceType;
+import com.android.util.Pair;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Simple implementation of {@link IProjectCallback}. This is a very basic implementation that
+ * cannot support custom classes. Look for the one in ADT for custom class support.
+ *
+ * Because there's no custom view support, the int to resource name is all dynamic instad of
+ * looking up in the R.java class that was compiled.
+ *
+ */
+public class ProjectCallback implements IProjectCallback {
+  private Map<ResourceType, Map<String, Integer>> mIdMap =
+      new HashMap<ResourceType, Map<String, Integer>>();
+
+  private Map<Integer, Pair<ResourceType, String>> mReverseIdMap =
+      new HashMap<Integer, Pair<ResourceType, String>>();
+
+  public ProjectCallback() {}
+
+  public AdapterBinding getAdapterBinding(
+      ResourceReference adapterViewRef, Object adapterCookie, Object viewObject) {
+    // TODO Auto-generated method stub
+    return null;
+  }
+
+  public Object getAdapterItemValue(ResourceReference adapterView, Object adapterCookie,
+      ResourceReference itemRef, int fullPosition, int positionPerType, int fullParentPosition,
+      int parentPositionPerType, ResourceReference viewRef, ViewAttribute viewAttribute,
+      Object defaultValue) {
+    // TODO Auto-generated method stub
+    return null;
+  }
+
+  public String getNamespace() {
+    // no custom class == no custom attribute, this is not needed.
+    return null;
+  }
+
+  public ILayoutPullParser getParser(String layoutName) {
+    // don't support custom parser for included files.
+    return null;
+  }
+
+  public ILayoutPullParser getParser(ResourceValue layoutResource) {
+    // don't support custom parser for included files.
+    return null;
+  }
+
+  public Integer getResourceId(ResourceType type, String name) {
+    // since we don't have access to compiled id, generate one on the fly.
+    Map<String, Integer> typeMap = mIdMap.get(type);
```

Comment:
We definitely should integrate LayoutLibrary assets with the ResourceLoader of robolectric and avoid loading from SDK folder once we are happy with the rendering.